src/solver/application.cpp

```diff
@@ -239,6 +249,8 @@ void Application::prepare(int argc, char* argv[])
 
     checkSimplexRangeHydroHeuristic(pParameters->simplexOptimizationRange, pStudy->areas);
 
+    checkAdqPatchStudyModeEconomuOnly(pParameters->include.adequacyPatch, pParameters->mode);
+
```

Comment:
Add a check here : if adq-patch is on and there is no area in the adq-patch : throw an error.